### ISSUE TITLE
[BUG] After using govc to add custom attributes to a VM within vcsim, govc lists duplicate entries (but PowerShell does not)

### ISSUE THREAD

**Issue by atc0005** [vmware/govmomi#2692]
## Describe the bug

After adding custom attributes and then setting custom attributes for a specific VM duplicate entries are shown via `govc fields.info`, but not via PowerShell snippet.

## To Reproduce

Overview:

1. Use `govc` to export inventory from live vCenter
2. Launch vcsim using exported inventory
3. List Custom Attributes to confirm current entries
4. Add new Custom Attributes
5. Set new Custom Attributes for a specific VM (in vcsim)
6. List Custom Attributes for VM
7. Note that there are two of each Custom Attribute just added

Details:

```shell
# Listing original CAs, setting new ones
govc fields.ls
govc fields.add -type VirtualMachine 'Backup Status'
govc fields.add -type VirtualMachine 'Last Backup'
govc fields.set 'Backup Status' 'Backup Job ID [43915] Client: [vcapps], Backup Set: [defaultBackupSet], Subclient: [VMs With Validation] Parent Backup Job ID [43901]' rhelcluster2
govc fields.set 'Last Backup' '2021-11-09 9:07:21 PM' rhelcluster2
govc fields.info rhelcluster2
```

```console
# List Custom Attributes via govc
$ govc fields.info rhelcluster2
/Example/host/VC1/esx2.example.com/rhelcluster2  Backup Status  Backup Job ID [43915] Client: [vcapps], Backup Set: [defaultBackupSet], Subclient: [VMs With Validation] Parent Backup Job ID [43901]
/Example/host/VC1/esx2.example.com/rhelcluster2  Backup Status  Backup Job ID [43915] Client: [vcapps], Backup Set: [defaultBackupSet], Subclient: [VMs With Validation] Parent Backup Job ID [43901]
/Example/host/VC1/esx2.example.com/rhelcluster2  Last Backup    2021-11-09 9:07:21 PM
/Example/host/VC1/esx2.example.com/rhelcluster2  Last Backup    2021-11-09 9:07:21 PM

$ govc fields.info -json rhelcluster2
{
  "Info": [
    {
      "Object": {
        "Type": "VirtualMachine",
        "Value": "vm-104773"
      },
      "Path": "/Example/host/VC1/esx2.example.com/rhelcluster2",
      "Name": "rhelcluster2",
      "Key": "Backup Status",
      "Value": "Backup Job ID [43915] Client: [vcapps], Backup Set: [defaultBackupSet], Subclient: [VMs With Validation] Parent Backup Job ID [43901]"
    },
    {
      "Object": {
        "Type": "VirtualMachine",
        "Value": "vm-104773"
      },
      "Path": "/Example/host/VC1/esx2.example.com/rhelcluster2",
      "Name": "rhelcluster2",
      "Key": "Backup Status",
      "Value": "Backup Job ID [43915] Client: [vcapps], Backup Set: [defaultBackupSet], Subclient: [VMs With Validation] Parent Backup Job ID [43901]"
    },
    {
      "Object": {
        "Type": "VirtualMachine",
        "Value": "vm-104773"
      },
      "Path": "/Example/host/VC1/esx2.example.com/rhelcluster2",
      "Name": "rhelcluster2",
      "Key": "Last Backup",
      "Value": "2021-11-09 9:07:21 PM"
    },
    {
      "Object": {
        "Type": "VirtualMachine",
        "Value": "vm-104773"
      },
      "Path": "/Example/host/VC1/esx2.example.com/rhelcluster2",
      "Name": "rhelcluster2",
      "Key": "Last Backup",
      "Value": "2021-11-09 9:07:21 PM"
    }
  ]
}
```

I had the duplicate entries even if using the fully-qualified inventory path for the VM with `govc fields.info`.

PowerShell v7.2.1 / PowerCLI v12.4.0 however didn't seem to have the same problem:

```powershell
# Attempt to replicate the duplicate Custom Attribute listings
Get-VM -Name rhelcluster2 | Select-Object -Property CustomFields

CustomFields
------------
{[Tier, ], [Backup Status, Backup Job ID [43915] Client: [vcapps], Backup Set: [defaultBackupSet], Subclient: [VMs With Validation] Parent Backup Job ID [43901]], [Last Backup, 2021-11-09 9:07:21 PM]}
```

## Expected behavior

I expected to see this output:

```console
# List Custom Attributes via govc
$ govc fields.info rhelcluster2
/Example/host/VC1/esx2.example.com/rhelcluster2  Backup Status  Backup Job ID [43915] Client: [vcapps], Backup Set: [defaultBackupSet], Subclient: [VMs With Validation] Parent Backup Job ID [43901]
/Example/host/VC1/esx2.example.com/rhelcluster2  Last Backup    2021-11-09 9:07:21 PM
```

## Affected version

Both binaries were downloaded from the releases page.

```console
$ govc version -l
Build Version: 0.27.2
Build Commit: f04d77d6
Build Date: 2021-11-23T21:22:18Z

$ vcsim version -l
Build Version: 0.27.2
Build Commit: f04d77d6
Build Date: 2021-11-23T21:22:18Z
```

**Comment by dougm** [comment]
Hey @atc0005 , looks like the problem is vcsim always appends, when it should first check if an entry with `Key` already exists and set if so: https://github.com/vmware/govmomi/blob/59582c4b15bd063898fa424101b10db1030dc463/simulator/custom_fields_manager.go#L193-L194

**Comment by atc0005** [comment]
> Hey @atc0005 , looks like the problem is vcsim always appends, when it should first check if an entry with `Key` already exists and set if so

Hey @dougm, thanks for the feedback.

Do you know if PowerCLI has de-duping behavior that prevents it from displaying duplicate entries like govc currently does?

Looking around, I see that @nilsanderselde encountered this previously and reported it on #2687. He proposed a fix here: https://github.com/nilsanderselde/govmomi/pull/1/files?diff=split&w=1

The differences in behavior between PowerCLI and govc led me to believe that I was encountering a separate problem, but perhaps not?

**Comment by dougm** [comment]
Not sure about PowerCLI, it might be unintentionally doing so, but neither client should need to de-dup.  I still have lots of issues to catch up on, had not looked at #2687 , but that would solve the problem here.

**Comment by dougm** [comment]
Fixed in #2700
